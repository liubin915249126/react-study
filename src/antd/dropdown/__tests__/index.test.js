import React from 'react';
import { mount } from 'enzyme';
import Dropdown from '..';
import Menu from '../../menu';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('Dropdown', () => {
  mountTest(() => (
    <Dropdown menu={<Menu />}>
      <span />
    </Dropdown>
  ));

  rtlTest(() => (
    <Dropdown menu={<Menu />}>
      <span />
    </Dropdown>
  ));

  it('overlay is function and has custom transitionName', () => {
    const wrapper = mount(
      <Dropdown overlay={() => <div>menu</div>} transitionName="move-up" visible>
        <button type="button">button</button>
      </Dropdown>,
    );
    expect(wrapper).toMatchRenderedSnapshot();
  });

  it('overlay is string', () => {
    const wrapper = mount(
      <Dropdown overlay="string" visible>
        <button type="button">button</button>
      </Dropdown>,
    );
    expect(wrapper).toMatchRenderedSnapshot();
  });
});
