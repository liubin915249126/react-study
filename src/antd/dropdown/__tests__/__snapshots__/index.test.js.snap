// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown overlay is function and has custom transitionName 1`] = `
Array [
  <button
    class="ant-dropdown-trigger"
    type="button"
  >
    button
  </button>,
  <div>
    <div
      class="ant-dropdown"
      style="opacity:0;pointer-events:none"
    >
      <div>
        menu
      </div>
    </div>
  </div>,
]
`;

exports[`Dropdown overlay is string 1`] = `
Array [
  <button
    class="ant-dropdown-trigger"
    type="button"
  >
    button
  </button>,
  <div>
    <div
      class="ant-dropdown"
      style="opacity:0;pointer-events:none"
    >
      <span>
        string
      </span>
    </div>
  </div>,
]
`;

exports[`Dropdown rtl render component should be rendered correctly in RTL direction 1`] = `
<span
  class="ant-dropdown-trigger ant-dropdown-rtl"
/>
`;
